include/envoy/config/context_provider.h

```diff
@@ -14,12 +16,45 @@ namespace Config {
  */
 class ContextProvider {
 public:
+  using UpdateCb = std::function<void(absl::string_view)>;
+
   virtual ~ContextProvider() = default;
 
   /**
-   * @return const xds::core::v3::ContextParams& node-level context parameters.
+   * @return const xds::core::v3::ContextParams& static node-level context parameters.
    */
   virtual const xds::core::v3::ContextParams& nodeContext() const PURE;
+
+  /**
+   * @param resource_type_url resource type URL for context parameters.
+   * @return const xds::core::v3::ContextParams& dynamic node-level context parameters.
+   */
+  virtual const xds::core::v3::ContextParams&
```

Comment:
since it's in node, I am not sure if we want to differentiate between different resource types. May I ask why the decision?
Likely we will have the same value for all the keys if we choose to do this.

Reply:
As mentioned offline, this same interface will be used for both `Node` and `xdstp://` (in a future PR). So, to future proof for that, we need to provide this granularity.